src/cps/search.cpp

```diff
@@ -71,36 +71,71 @@ namespace cps::search {
             return out;
         }
 
+        fs::path libdir() {
+            // TODO: libdir needs to be configurable based on the personality,
+            //       and different name schemes.
+            //       This is complicated by the fact that different distros have
+            //       different schemes.
+            return "lib";
+        }
+
+        fs::path datadir() {
+            // TODO: needs to be configurable. see libdir above.
+            return "share";
+        }
+
         const std::vector<fs::path> nix{"/usr", "/usr/local"};
         // TODO: const std::vector<std::string> mac{""};
         // TODO: const std::vector<std::string> win{""};
 
         std::vector<fs::path> cached_paths{};
 
-        const std::vector<fs::path> search_paths(Env env) {
+        /// @brief expands a single search prefix into a set of full paths
+        /// @param env stored environment variables
+        /// @return A vector of paths to search, in order
+        std::vector<fs::path> expand_prefix(const fs::path & path) {
+            std::vector<fs::path> paths{};
+
+            // TODO: macOS specific paths
+            // TODO: Windows specific paths
```

Comment:
I think potential contributors would benefit from "help wanted" issues for relatively straightforward tasks. Should we make issues for these? We can keep the TODOs in the code and referencing the issues if there's benefit to that.